src/Xamarin.Android.Build.Tasks/Tasks/ResolveAssemblies.cs

```diff
@@ -90,7 +90,7 @@ void Execute (MetadataResolver resolver)
 						if (lockFile != null)
 							resolved_assembly = ResolveRuntimeAssemblyForReferenceAssembly (lockFile, assembly.ItemSpec);
 						if (lockFile == null || resolved_assembly == null) {
-							LogCodedWarning ("XA0107", resolved_assembly, 0, "Ignoring {0} as it is a Reference Assembly", resolved_assembly);
+							LogCodedWarning ("XA0107", resolved_assembly, 0, "Ignoring {0} as it is a Reference Assembly", resolved_assembly ?? assembly.ItemSpec);
```

Comment:
Close, but not quite; if `resolved_assembly` is `null`, which is what we're assuming will cause the assembly name to be empty, then that means that the `file` parameter provide to `LogCodedWarning()` will *also* be null/empty, which presumably is not quite useful within the IDE.